WL-3242 Add a reset poll button

nicholaswilson100 opened the pull request:
I :heart: RSF

nicholaswilson100 committed:
WL-3242 Add a reset poll button

Puts the reset button at the bottom of the page, next delete. Allows
multiple polls to be reset.

nicholaswilson100 committed:
WL-3242 Fix Javadoc, indentation and security checks

buckett commented on api/src/java/org/sakaiproject/poll/logic/ExternalLogic.java:
```diff
@@ -232,4 +232,13 @@
+     * 	The owner of the poll
+     * @return true or false
+     */
+    public boolean userCanDeletePoll(String owner);
```
I think ExternalLogic is designed to be the wrapper around all the Sakai services so that the polls tool doesn't bind against 10 Sakai services but just binds against the ExternalLogic interfaces instead.

This doesn't seem like the right place for polls business logic, PollListManager looks to be the better place.

nicholaswilson100 committed:
WL-3242 Move userCanDeletePoll method

Moving userCanDeletetPoll method from ExternalLogic.java to PollListManager.java

buckett commented on tool/src/java/org/sakaiproject/poll/tool/params/PollToolBean.java:
```diff
@@ -206,6 +206,22 @@ public void processActionDelete() {
+                try {
+                    pollVoteManager.deleteAll(votes);
+                }
+                catch(SecurityException e){
```
Is this redundant as I'm not sure it gets thrown.

nicholaswilson100 committed:
WL-3242 Fix indentation in PollToolBean

nicholaswilson100 committed:
WL-3242 Removing unecessary catch

nicholaswilson100 commented:
I've removed the unnecessary exception

nicholaswilson100 commented:
I created a branch 'WL-3242' in my Sakai 8 checkout and added, committed and pushed the same changes there.
